# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/parted/hotfix-atari-locale.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- parted-3.6/libparted/labels/atari.c.vanilla	2023-07-18 20:12:39.843681000 +0500
+++ parted-3.6/libparted/labels/atari.c	2023-07-18 20:14:18.235681000 +0500
@@ -1962,7 +1962,8 @@
 	PED_ASSERT (sizeof (AtariRawPartition) == 12);
 	PED_ASSERT (sizeof (AtariRawTable) == 512);
 	/* GNU Libc doesn't support NULL instead of the locale name */
-	PED_ASSERT ((atr_c_locale = newlocale(LC_ALL_MASK, "C", NULL)) != NULL);
+	atr_c_locale = newlocale(LC_ALL_MASK, "C", NULL);
+	PED_ASSERT (atr_c_locale != NULL);
 
 	ped_disk_type_register (&atari_disk_type);
 }
